src/main/java/com/marginallyclever/convenience/ButtonIcon.java

```diff
@@ -0,0 +1,37 @@
+package com.marginallyclever.convenience;
+
+import com.marginallyclever.makelangelo.Translator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class ButtonIcon extends JButton {
+    private static final Logger logger = LoggerFactory.getLogger(ButtonIcon.class);
+
+    public ButtonIcon(String translationKey, String iconPath) {
+        super();
+        if (translationKey != null && !translationKey.isEmpty() && !translationKey.isBlank()) {
```

Comment:
Does Translator.get() fail if given a null?  That might be the place to return "no key provided" or something equivalent.

Reply:
it throws a NullPointerException if I remember correctly. I suggest improving Translator in an other PR

Reply:
What was the other PR #?

Reply:
it is not yet developed